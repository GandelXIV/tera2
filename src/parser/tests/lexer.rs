use crate::parser::lexer2::tokenize;

#[test]
fn test_lexer_ok() {
    insta::glob!("lexer_inputs/success/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let tokens: Result<Vec<_>, _> = tokenize(&contents).collect();
        let tokens = tokens.unwrap().into_iter().map(|x| x.0).collect::<Vec<_>>();
        insta::assert_debug_snapshot!(&tokens);
    });
}

#[test]
fn test_lexer_errors() {
    insta::glob!("lexer_inputs/errors/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let res: Result<Vec<_>, _> = tokenize(&contents).collect();
        assert!(res.is_err());
        insta::assert_debug_snapshot!(res.unwrap_err());
    });
}
