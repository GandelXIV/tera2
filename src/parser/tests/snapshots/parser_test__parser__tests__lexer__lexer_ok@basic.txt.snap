---
source: src/parser/tests/lexer.rs
assertion_line: 9
expression: "&tokens"
input_file: src/parser/tests/lexer_inputs/success/basic.txt

---
[
    CONTENT("Hello "),
    VARIABLE_START(false),
    IDENT(username),
    VARIABLE_END(false),
    CONTENT(", {welcome}."),
]
